src/algorithms/wolfe_line_search.h

```diff
@@ -0,0 +1,118 @@
+#pragma once
+
+#include <RcppArmadillo.h>
+#include "../families/family.h"
+
+using namespace Rcpp;
+using namespace arma;
+
+template <typename T>
+double Family::zoom(const T& x, const mat&y, const double rho, const mat&u, 
+                    const mat& z, const mat& d, double t_low, double t_high){
+  
+  mat lin_pred = x*z;
+  const double f0 = primal(y,lin_pred)+0.5*rho*pow(norm(z-u),2);
+  const mat g0 = gradient(x,y,lin_pred)+rho*(z-u);
+  const double dec0 = dot(g0,d);
+
+  const double c1 = 1e-4;
+  const double c2 = 0.9;
+
+  uword max_iter = 20;
+  uword iter = 0;
+
+  while(true){
+
+    double f_low = primal(y,x*(z+t_low*d)) + 
+                   0.5*rho*pow(norm(z+t_low*d-u),2);
+
+    double t_mid = 0.5*(t_low+t_high);
+
+    mat z_mid = z+t_mid*d;
+
+    mat lin_pred = x*z_mid;
+    double f_mid = primal(y,lin_pred) + 
+                   0.5*rho*pow(norm(z_mid-u),2);
+
+    if((f_mid > f0 + c1*t_mid*dec0) || (f_mid >= f_low)){
+      t_high = t_mid;
+    }
+    else{
```

Comment:
Put brackets for else statements on one line.
```suggestion
    } else {
```